cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge_v2_outbound.rs

```diff
@@ -454,6 +454,92 @@ fn transact_with_agent_from_asset_hub() {
 	});
 }
 
+#[test]
+fn transact_with_agent_from_asset_hub_without_any_asset_transfer() {
+	fund_on_bh();
+
+	register_assets_on_ah();
+
+	fund_on_ah();
+
+	AssetHubWestend::execute_with(|| {
+		type RuntimeOrigin = <AssetHubWestend as Chain>::RuntimeOrigin;
+
+		let local_fee_asset =
+			Asset { id: AssetId(Location::parent()), fun: Fungible(LOCAL_FEE_AMOUNT_IN_DOT) };
+
+		let remote_fee_asset =
+			Asset { id: AssetId(ethereum()), fun: Fungible(REMOTE_FEE_AMOUNT_IN_ETHER) };
+
+		let assets = vec![local_fee_asset.clone(), remote_fee_asset.clone()];
+
+		let beneficiary =
+			Location::new(0, [AccountKey20 { network: None, key: AGENT_ADDRESS.into() }]);
+
+		let transact_info =
+			ContractCall::V1 { target: Default::default(), calldata: vec![], gas: 40000, value: 0 };
+
+		let xcms = VersionedXcm::from(Xcm(vec![
+			WithdrawAsset(assets.clone().into()),
+			PayFees { asset: local_fee_asset.clone() },
+			InitiateTransfer {
+				destination: ethereum(),
+				remote_fees: Some(AssetTransferFilter::ReserveWithdraw(Definite(
+					remote_fee_asset.clone().into(),
+				))),
+				preserve_origin: true,
+				assets: vec![AssetTransferFilter::ReserveWithdraw(Wild(AllOf {
+					id: AssetId(ethereum()),
+					fun: WildFungibility::Fungible,
+				}))],
```

Comment:
Actually, there are no reserved assets left in this case. Check the log on BH for demonstration - there is a `WithdrawAsset(Assets([]))` which means `ENA` is a empty vec, but not none.

```
WithdrawAsset(ETH)
PayFees(ETH)
WithdrawAsset(Assets([]))
AliasOrigin(Origin)
DepositAsset(Asset)
Transact()
SetTopic(Topic)
```

Reply:
it fails with:

```
ERROR xcm::ethereum_blob_exporter::v2: unroutable due to pattern matching error 'NoReserveAssets'.
```

Reply:
(not a blocker, dry-running will catch this, but could be improved)

Reply:
Nice catch! Fixed in https://github.com/vgeddes/polkadot-sdk/pull/59/commits/ac515e3a04f972c51d92343bf3af783eec2a24f5